hpct-ldap-server/src/servers/ldap.py

```diff
@@ -0,0 +1,503 @@
+
+                rc = subprocess.call(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+                if rc != 0:
+                    raise Exception(
+                        f'Unable to add group. Make sure to run "ldapadd -x -D cn=admin,{dcs} -W -f /etc/ldap/basedn.ldif" first.'
+                    )
+
+    def disable(self) -> None:
+        """Disable services."""
+        for name in self.systemd_services:
+            systemd.service_pause(name)
+
+    def enable(self) -> None:
+        """Enable services."""
+        for name in self.systemd_services:
+            systemd.service_resume(name)
+
+    def install(self) -> None:
+        """Install packages."""
+        if self.packages:
+            try:
+                apt.update()
+                for name in self.packages:
+                    apt.add_package(name)
+            except:
+                raise Exception(f"failed to install package ({name})")
+
+    def is_enabled(self) -> bool:
+        """Check enabled status of services."""
+        if self.systemd_services:
+            for name in self.systemd_services:
+                if not systemd._systemctl("is-enabled", name, quiet=True):
+                    return False
+
+        return True
+
+    def is_installed(self) -> bool:
+        """Check packages are installed."""
+        if self.packages:
+            for name in self.packages:
+                if not apt.DebianPackage.from_installed_package(name).present:
+                    return False
+
+        return True
+
+    def is_running(self) -> bool:
+        """Check running/active status of services."""
+        if self.systemd_services:
+            for name in self.systemd_services:
+                if not systemd.service_running(name):
+                    return False
+
+        return True
+
+    def restart(self) -> None:
+        """Restart servers/services."""
+        self.stop()
+        self.start()
+
+    def set_config(self, admin_passwd=None, domain=None, org=None) -> None:
+        """Set LDAP password and slapd configuration.
+
+        Parameters
+        ----------
+        admin_passwd : str
+            LDAP password.
+        domain : str
+            Domain name.
+        org : str
+            Organization name.
+        """
+        if domain and org and admin_passwd:
+            dcs = self._split_domain(domain)
+
+            # Reconfigure slapd with password
+            self.tls_deb(admin_passwd, domain, org)
+            self._add_base(admin_passwd, dcs)
+        else:
+            raise Exception("Domain, Organization, or Password is missing cannot complete action.")
+
+    def start(self) -> None:
+        """Start services."""
+        for name in self.systemd_services:
+            systemd.service_start(name)
+
+    def stop(self) -> None:
+        """Stop services."""
+        for name in self.systemd_services:
+            systemd.service_stop(name)
+
+    def tls_deb(self, admin_passwd, domain, org) -> None:
```

Comment:
Is there a better name than `tls_deb`? It is not clear the role of "tls" for this.

Reply:
Updated to `slapd_config` since it is setting the debconf configuration selections for the slapd package.